dev/ColorPicker/readme.md

```diff
@@ -0,0 +1,141 @@
+* [Color Picker Class Docs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.colorpicker) Useful as an API reference for developers implementing this control in an application.
+
+The ColorPicker directory contains a set of controls with the primary control being the `ColorPicker` itself. These controls, their relationships, and function are outlined below and then described in the following sections.
+
+1. `ColorPicker` : The top-level control containing all other controls. This is the control intended for user interaction and used by application developers.
+2. `ColorPickerSlider` : A custom slider implementation used for the sliders in the `ColorPicker` control template itself. These sliders are unique in that they have a gradient background rendered to correspond with a defined HSV color channel. Sliders are 1-dimensional. This control is not useful apart from the `ColorPicker`.
+3. `ColorSpectrum` : The main graphic displaying a field of possible colors the user can select from. The color spectrum is 2-dimensional and can be viewed in a Box or Ring shape.
+4. `SpectrumBrush` : A special brush used to render the color spectrum background using composition APIs. This is only available on Windows version 1703 (RS2) and later.
+
+The control hierarchy and namespaces are as follows:
+
+```
+ ─ ColorPicker             [Microsoft.UI.Xaml.Controls]
+   ├── ColorPickerSlider   [Microsoft.UI.Xaml.Controls.Primitives]
+   └── ColorSpectrum       [Microsoft.UI.Xaml.Controls.Primitives]
+       └── SpectrumBrush   [Microsoft.UI.Xaml.Controls.Primitives]
+```
+
+Generalized helper methods are also located in two places:
+
+ * ColorHelpers.[cpp/h] : Helpers specifically for the color picker and related controls. These are located in the `dev/ColorPicker` directory itself.
+ * ColorConversion.[cpp/h] : More generalized helpers for working with RGB and HSV colors. The RGB and HSV structures are defined here as well as conversions between the two color representations and hex. These files are located in the `dev/Common` directory separate from ColorPicker.
+
+## ColorPicker
+
+[Control Properties](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.colorpicker#properties), [Control Events](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.colorpicker#events)
+
+Most of the detailed functionality specific to the `ColorPicker` is implemented in the primitive controls outlined below. See the links above and at the top of this document for full details on the usage of the `ColorPicker` itself.
+
+The `ColorPicker` functions by monitoring for changing in properties or user selection and then updating the selected color and graphical controls (preview rectangles, sliders, input text boxes, etc.) accordingly. It also ensures the various properties and input is validated before computing the updated selected color.
```

Comment:
So when I click on the colorspectrum what happens after that? How do changes in properties of the subcomponents get reflected in the overall control?

Reply:
The color spectrum is linked to the color picker using the Vector4 HsvColor property and the ColorChanged event. The other properties are going ColorPicker -> ColorSpectrum and are simply set in the style. I would encourage anyone to read the code for this level of information.

Reply:
Added a note to clarify this section regardless